lit_tests/kernel/wave/codegen.py

```diff
@@ -1930,6 +1930,65 @@ def broadcast_add(
     # CHECK: arith.addf %[[LHS_1]], %[[BCAST_RHS_1]] : vector<2xf16>
 
 
+@run_test
+def test_broadcast_scaled_add():
+    M = tkl.sym.M
+    N = tkl.sym.N
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+    shape = (256, 256)
+
+    wave_size = 64
+    BLOCK_M = 1
+    BLOCK_N = sympy.Max(sympy.Min(shape[1], 256), wave_size)
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=wave_size,
+            vector_shapes={M: BLOCK_M, N: BLOCK_N},
+        )
+    ]
+
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    @tkw.wave(constraints)
```

Comment:
Can we add these kernels to templates to avoid duplicating them?

Reply:
Probably need your guidance on this one, I was thinking of reserving templates for kernels that are actually useful for end users not unlike https://github.com/triton-lang/triton/tree/main/python/triton_kernels/triton_kernels. 

Since these kernels sole purpose right now is for testing compiler feature I was thinking of keeping them in the test itself. I don't mind either ways, but would love your thoughts on it. :)

Reply:
Maybe we can have a test templates for things that are not our production kernels, but that keep getting reused for tests etc. I think we have a lot of code duplication because of this.

Reply:
makes sense, I will move for now, and then factor away the performant/deployment kernels in future PR